csrc/torch_memory_saver.cpp

```diff
@@ -177,16 +191,30 @@ class TorchMemorySaver {
     void pause(const std::string& tag) {
         const std::lock_guard <std::mutex> lock(allocator_metadata_mutex_);
 
+        bool found_valid_allocation = false;
         for (auto it = allocation_metadata_.begin(); it != allocation_metadata_.end(); ++it) {
             void *ptr = it->first;
-            _AllocationMetadata metadata = it->second;
+            _AllocationMetadata &metadata = it->second;
 
             if (!tag.empty() && metadata.tag != tag) {
                 continue;
             }
 
+            if (metadata.state == AllocationState::PAUSED) {
+                std::string error_msg = "Trying to pause already paused allocation";
+                if (!tag.empty()) {
+                    error_msg += " for tag: " + tag;
+                }
+                error_msg += " ptr=" + std::to_string(reinterpret_cast<uintptr_t>(ptr));
+                std::cerr << "[torch_memory_saver.cpp] ERROR: " << error_msg << std::endl;
```

Comment:
similar about this (and other things), maybe just a `std::cerr << ...` and no std::string msg